wrapper/src/main/java/software/amazon/jdbc/plugin/federatedauth/FederatedAuthPlugin.java

```diff
@@ -0,0 +1,317 @@
+
+  private static final String TELEMETRY_FETCH_TOKEN = "fetch IAM token";
+  private static final Logger LOGGER = Logger.getLogger(FederatedAuthPlugin.class.getName());
+
+  protected final PluginService pluginService;
+
+  protected final RdsUtils rdsUtils = new RdsUtils();
+
+  private static final Set<String> subscribedMethods =
+      Collections.unmodifiableSet(new HashSet<String>() {
+        {
+          add("connect");
+          add("forceConnect");
+        }
+      });
+
+  static {
+    PropertyDefinition.registerPluginProperties(FederatedAuthPlugin.class);
+  }
+
+  private final TelemetryFactory telemetryFactory;
+  private final TelemetryGauge cacheSizeGauge;
+  private final TelemetryCounter fetchTokenCounter;
+
+  @Override
+  public Set<String> getSubscribedMethods() {
+    return subscribedMethods;
+  }
+
+  public FederatedAuthPlugin(final PluginService pluginService,
+                             final CredentialsProviderFactory credentialsProviderFactory) {
+    try {
+      Class.forName("software.amazon.awssdk.services.sts.model.AssumeRoleWithSamlRequest");
+    } catch (final ClassNotFoundException e) {
+      throw new RuntimeException(Messages.get("FederatedAuthPlugin.javaStsSdkNotInClasspath"));
+    }
+    this.pluginService = pluginService;
+    this.credentialsProviderFactory = credentialsProviderFactory;
+    this.telemetryFactory = pluginService.getTelemetryFactory();
+    this.cacheSizeGauge = telemetryFactory.createGauge("federatedAuth.tokenCache.size", () -> (long) tokenCache.size());
+    this.fetchTokenCounter = telemetryFactory.createCounter("federatedAuth.fetchToken.count");
+  }
+
+
+  @Override
+  public Connection connect(
+      final String driverProtocol,
+      final HostSpec hostSpec,
+      final Properties props,
+      final boolean isInitialConnection,
+      final JdbcCallable<Connection, SQLException> connectFunc)
+      throws SQLException {
+    return connectInternal(hostSpec, props, connectFunc);
+  }
+
+  @Override
+  public Connection forceConnect(
+      final @NonNull String driverProtocol,
+      final @NonNull HostSpec hostSpec,
+      final @NonNull Properties props,
+      final boolean isInitialConnection,
+      final @NonNull JdbcCallable<Connection, SQLException> forceConnectFunc)
+      throws SQLException {
+    return connectInternal(hostSpec, props, forceConnectFunc);
+  }
+
+  private Connection connectInternal(HostSpec hostSpec, Properties props,
+      JdbcCallable<Connection, SQLException> connectFunc) throws SQLException {
+
+    checkIdpCredentialsWithFallback(props);
+
+    String host = IamAuthUtils.getIamHost(IAM_HOST.getString(props), hostSpec);
+
+    int port = IamAuthUtils.getIamPort(
+        StringUtils.isNullOrEmpty(IAM_DEFAULT_PORT.getString(props)) ? -1 : IAM_DEFAULT_PORT.getInteger(props),
+        hostSpec,
+        this.pluginService.getDialect().getDefaultPort());
```

Comment:
```suggestion
    int port = IamAuthUtils.getIamPort(
        IAM_DEFAULT_PORT.getInteger(props),
        hostSpec,
        this.pluginService.getDialect().getDefaultPort());
```

And set the default value for the IAM_DEFAULT_PORT to -1?